packages/ui/src/components/va-skeleton/VaSkeleton.vue

```diff
@@ -0,0 +1,210 @@
+<template>
+  <component class="va-skeleton" :class="classes" :is="tag">
+    <slot />
+    <div class="va-skeleton-wave" v-if="animation === 'wave'" />
+  </component>
+</template>
+
+<script lang="ts">
+import { defineComponent, onMounted, ref, computed, PropType } from 'vue'
+import { useBem, useColors } from '../../composables'
+
+export default defineComponent({
+  name: 'VaSkeleton',
+
+  inheritAttrs: true,
+
+  props: {
+    color: { type: String, default: 'backgroundElement' },
+    delay: { type: Number, default: 100 },
+
+    tag: { type: String, default: 'div' },
+    inline: { type: Boolean, default: false },
+
+    animation: { type: String as PropType<'pulse' | 'wave' | 'none'>, default: 'pulse' },
+
+    lines: { type: Number, default: 1 },
+    height: { type: [Number, String], default: '1em' },
+    width: { type: [String], default: '100%' },
+    lineGap: { type: String, default: '8px' },
+    textWidth: { type: [String], default: '75%' },
+
+    variant: { type: String as PropType<'text' | 'circle' | 'rounded' | 'squared'>, default: 'squared' },
+  },
+
+  emits: [/* Define emits here */],
+
+  setup (props, { attrs }) {
+    const doShow = ref(false)
+
+    onMounted(() => {
+      setTimeout(() => {
+        doShow.value = true
+      }, props.delay)
+    })
+
+    const heightComputed = computed(() => {
+      if (props.variant === 'text') {
+        return `${props.lines}em`
+      }
+
+      return props.height
+    })
+    const widthComputed = computed(() => {
+      if (props.variant === 'circle') {
+        return heightComputed.value
+      }
+
+      if (typeof props.width === 'number') {
+        return `calc(100% * ${props.width})`
+      }
+
+      return props.width
+    })
+
+    const { getColor } = useColors()
+
+    const colorComputed = computed(() => getColor(props.color))
+    const negativeLineGap = computed(() => `-${props.lineGap}`)
+
+    const bem = useBem('va-skeleton', () => ({
+      lines: props.lines > 1,
+      text: props.variant === 'text',
+      circle: props.variant === 'circle',
+      hidden: !doShow.value,
+      inline: props.inline,
+      pulse: props.animation === 'pulse',
+      wave: props.animation === 'wave',
+    }))
+
+    const borderRadius = computed(() => {
+      if (props.variant === 'circle') { return '50%' }
+      if (props.variant === 'rounded') { return `calc(${heightComputed.value} / 5)` }
+
+      return '0px'
+    })
```

Comment:
I'll suggest to use css variables here.

Reply:
There are sometimes when you want ot have rounded button and squared image for example. It will be a pain to deal with CSS variables for each individual case.